#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: Fix libstdc++ type definitions


dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- gcc/libstdc++-v3/config/locale/uclibc/monetary_members.cc.uclibc200_wchar~	2006-03-10 15:32:37 +0100
+++ gcc/libstdc++-v3/config/locale/uclibc/monetary_members.cc	2006-03-10 15:37:27 +0100
@@ -401,7 +401,7 @@
 # ifdef __UCLIBC_HAS_XLOCALE__
 	  _M_data->_M_decimal_point = __cloc->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __cloc->thousands_sep_wc;
-# else
+# elif defined __UCLIBC_HAS_LOCALE__
 	  _M_data->_M_decimal_point = __global_locale->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __global_locale->thousands_sep_wc;
 # endif
@@ -556,7 +556,7 @@
 # ifdef __UCLIBC_HAS_XLOCALE__
 	  _M_data->_M_decimal_point = __cloc->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __cloc->thousands_sep_wc;
-# else
+# elif defined __UCLIBC_HAS_LOCALE__
 	  _M_data->_M_decimal_point = __global_locale->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __global_locale->thousands_sep_wc;
 # endif
--- gcc/libstdc++-v3/config/locale/uclibc/numeric_members.cc.orig	2006-04-13 18:16:01.000000000 +0200
+++ gcc/libstdc++-v3/config/locale/uclibc/numeric_members.cc	2006-04-13 18:23:00.000000000 +0200
@@ -137,7 +137,7 @@
 # ifdef __UCLIBC_HAS_XLOCALE__
 	  _M_data->_M_decimal_point = __cloc->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __cloc->thousands_sep_wc;
-# else
+# elif __UCLIBC_HAS_LOCALE__
 	  _M_data->_M_decimal_point = __global_locale->decimal_point_wc;
 	  _M_data->_M_thousands_sep = __global_locale->thousands_sep_wc;
 # endif
